src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -892,9 +900,11 @@ variableDeclarator![AST mods, AST t]
         {#variableDeclarator = #(#[VARIABLE_DEF,"VARIABLE_DEF"], mods, #(#[TYPE,"TYPE"],d), id, v);}
     ;
 
-declaratorBrackets[AST typ]
-    :    {#declaratorBrackets=typ;}
-        (lb:LBRACK^ {#lb.setType(ARRAY_DECLARATOR);} RBRACK)*
+declaratorBrackets![AST typ]
+    :    ({LA(1) == AT}? an:annotations | ) lb:LBRACK {#lb.setType(ARRAY_DECLARATOR);} rb:RBRACK
+         db:declaratorBrackets[#(lb, typ, an, rb)]
+        {#declaratorBrackets = #db;}
+    |   {#declaratorBrackets = typ;}
```

Comment:
If this rule produces the AST we want, is there a better way to write it?

I think the issue with what I had before, 
```
({LA(1) == AT}? annotations | ) lb:LBRACK^ {#lb.setType(ARRAY_DECLARATOR);} RBRACK)*
```
is that for `@Nullable [] @Nullable []` it creates the first tree, for the first `@Nullable []` pair, then it adds the second `@Nullable` as a child of that tree (a second ANNOTATIONS child under the first ARRAY_DECLARATOR), then it creates the second ARRAY_DECLARATOR as a new root and adds the previous tree as a child.

In this new version of the rule, I replaced the `*` with an explicit recursion, passing the first `@Nullable []` tree as a new `typ` node to use when building the second `@Nullable []` tree.